addons/mail/static/src/model/store.js

```diff
@@ -19,10 +19,29 @@ export class Store extends Record {
         return this.recordByLocalId.get(localId);
     }
 
+    onBusinessError(err) {
+        this._.FC_QUEUE.clear();
+        this._.FS_QUEUE.clear();
+        this._.FA_QUEUE.clear();
+        this._.FD_QUEUE.clear();
+        this._.FU_QUEUE.clear();
+        this._.RO_QUEUE.clear();
+        this._.RD_QUEUE.clear();
+        this._.RHD_QUEUE.clear();
+        this._.UPDATE = 0;
+        console.warn("store data insert partially aborted due to following error:", err);
+        throw err;
```

Comment:
I'm still not convinced by the need to do effort to keep half broken state though.

User won't see this warning and if the exception is caught due to bad business code it will still be caught.

If we really need a custom warning maybe at least it should be the actual error dialog?

Anyway I think the only valid strategy is to fix bad business code, and then it becomes a non-issue.